spark/dl/src/main/resources/serialization/bigdl.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+package serialization;
+message BigDLModule
+{
+    string name = 1; //module name
+    repeated BigDLModule subModules = 2; // sub modules
+    BigDLTensor weight = 3; // weight for each layer, serialized data are stored as either float or double
+    BigDLTensor bias = 4; // bias for each layer
+    repeated string preModules = 5; //pre modules of the same hierarchy
```

Comment:
there's an assumption that the layer name is unique. But currently, we can't guarantee this in code level. Do we check this before we serialize the model?